modules/setting/config_provider.go

```diff
@@ -40,12 +39,6 @@ type iniFileConfigProvider struct {
 	newFile bool // whether the file has not existed previously
 }
 
-// NewEmptyConfigProvider create a new empty config provider
-func NewEmptyConfigProvider() ConfigProvider {
```

Comment:
`NewEmptyConfigProvider` seems pretty easy to use, no need to check `err` again and again.

Reply:
Normally I'd agree.
However, seeing that it only changes the test files I'm rather against ignoring errors happening during config setting.

Reply:
Why an "empty" config would fail? Just like an empty slice or empty map. I do not think this "err" makes sense.

Reply:
Like @delvh said, this function was used only by testing code and the wrap is small. I couldn't see the reason to keep a simple wrap function on setting package for testing purpose.

Reply:
And for easy usage, I ignored all error if the setting content is empty.